scripts/GetClrDbg.sh

```diff
@@ -76,97 +103,238 @@ generate_nuget_config()
+    # Add new version constants here
+    # 'latest' version may be updated
+    # all other version contstants i.e. 'vs2015u2' may not be updated after they are finalized
+    version_string="$(echo $1 | awk '{print tolower($0)}')"
+    case $version_string in
+        latest)
+            __ClrDbgVersion=14.0.25520-preview-3139256
+            ;;
+        vs2015u2)
+            __ClrDbgVersion=14.0.25520-preview-3139256 #This version is now locked and should not be updated.
+            ;;
+        *)
+            simpleVersionRegex="^[0-9].*"
+            if ! [[ "$1" =~ $simpleVersionRegex ]]; then
+                echo "Error: '$1' does not look like a valid version number."
+                exit 1
+            fi
+            __ClrDbgVersion=$1
+            ;;
+    esac
+}
+
+# Prepares installation directory.
+prepare_install_location()
+{
+    if [ -z $__InstallLocation ]; then
+        echo "Error: Install location is not set"
+        exit 1
+    fi
+
+    if [ -f "$__InstallLocation" ]; then
         echo "Error: Path '$2' points to a regular file and not a directory"
         exit 1
-    elif [ ! -d "$2" ]; then
+    elif [ ! -d "$__InstallLocation" ]; then
         echo 'Info: Creating install directory'
-        mkdir -p $2
+        mkdir -p $__InstallLocation
         if [ "$?" -ne 0 ]; then
             echo "Error: Unable to create install directory: '$2'"
             exit 1
         fi
+    fi   
+}
+
+# Removes installation directory if remove option is specified.
+process_removal()
+{
+    if [ "$__RemoveExisting" = true ]; then
+        echo "Info: Attempting to remove '$__InstallLocation'"
+        if [ -d $__InstallLocation ]; then
+            lsof $__InstallLocation > /dev/null 2>&1
+            if [ "$?" -eq 0 ]; then
+                echo "Error: files are being used in location '$__InstallLocation'"
+                exit 1
+            fi
+
+            rm -rf $__InstallLocation
+            if [ "$?" -ne 0 ]; then
+                echo "Error: files could not be removed from '$__InstallLocation'"
+                exit 1
+            fi
+        fi
     fi
-    __InstallPath=$2
-fi
+}
+
+# Checks if the existing copy is the latest version.
+check_latest()
+{
+    __SuccessFile="$__InstallLocation/success.txt" 
+    if [ -f "$__SuccessFile" ]; then
+        __LastInstalled=$(cat "$__SuccessFile")
+        echo "Info: LastIntalled version of clrdbg is '$__LastInstalled'"
+        if [ "$__ClrDbgVersion"="__LastInstalled" ]; then
+            __SkipDownloads=true
+            echo "Info: ClrDbg is upto date"
+        fi
+    else
+        echo "Info: Previous installation at "$__InstallLocation" not found"
+    fi
+}
 
-pushd $__InstallPath > /dev/null 2>&1
-if [ "$?" -ne 0 ]; then
-    echo "Error: Unable to cd to install directory '$__InstallPath'"
+if [ -z "$1" ]; then
+    print_help
     exit 1
+else
+    parse_and_get_arguments $@
+
+    if [ -z $__ClrDbgMetaVersion ]; then
```

Comment:
Maybe I am missing something, but this seems like the wrong check to me. It seems like we really want something like -- the first argument that doesn't start with '-' is the version, the second is the install dir.